### PR TITLE
Drop go 1.12 and fix some things

### LINKED ISSUE #349: Use Go 1.13 trimpath flag
When we're to require Go 1.13 (not today) it makes sense to use the `-trimpath` flag to make builds more reproducible:
> The new go build flag -trimpath removes all file system paths from the compiled executable, to improve build reproducibility. 

https://golang.org/doc/go1.13

### PR DESCRIPTION
**Pull request by roman-khimov**
We no longer need to support Go 1.12, so we can fix #349 and #350. At the same time I've noticed some other minor problems, so they were also fixed (albeit it makes this PR somewhat unfocused).

### CODE REVIEW

File: pkg/smartcontract/param_type.go [hunk 1]
```diff
@@ -160,7 +161,7 @@ func ParseParamType(typ string) (ParamType, error) {
 	case "any":
 		return AnyType, nil
 	default:
-		return UnknownType, errors.Errorf("Unknown contract parameter type: %s", typ)
+		return UnknownType, fmt.Errorf("Unknown contract parameter type: %s", typ)
```
Comment (AnnaShaleva) [inline pkg/smartcontract/param_type.go:164]: Previous errors start with lowercase letter, although it's not so important.

File: pkg/smartcontract/parameter.go [hunk 2]
```diff
@@ -94,7 +94,7 @@ func (p Parameter) MarshalJSON() ([]byte, error) {
 	case InteropInterfaceType, AnyType:
 		resultRawValue = nil
 	default:
-		resultErr = errors.Errorf("Marshaller for type %s not implemented", p.Type)
+		resultErr = fmt.Errorf("Marshaller for type %s not implemented", p.Type)
```
Comment (AnnaShaleva) [inline pkg/smartcontract/parameter.go:97]: First capital letter.

File: pkg/smartcontract/parameter.go [hunk 3]
```diff
@@ -183,7 +183,7 @@ func (p *Parameter) UnmarshalJSON(data []byte) (err error) {
 		// stub, ignore value, it can only be null
 		p.Value = nil
 	default:
-		return errors.Errorf("Unmarshaller for type %s not implemented", p.Type)
+		return fmt.Errorf("Unmarshaller for type %s not implemented", p.Type)
```
Comment (AnnaShaleva) [inline pkg/smartcontract/parameter.go:186]: First capital letter.

File: pkg/smartcontract/parameter.go [hunk 4]
```diff
@@ -412,7 +412,7 @@ func NewParameterFromString(in string) (*Parameter, error) {
 			}
 			// We currently do not support following types:
 			if res.Type == ArrayType || res.Type == MapType || res.Type == InteropInterfaceType || res.Type == VoidType {
-				return nil, errors.Errorf("Unsupported contract parameter type: %s", res.Type)
+				return nil, fmt.Errorf("Unsupported contract parameter type: %s", res.Type)
```
Comment (AnnaShaleva) [inline pkg/smartcontract/parameter.go:415]: And here.

File: pkg/core/native/policy.go [hunk 5]
```diff
@@ -516,12 +516,19 @@ func (p *Policy) CheckPolicy(d dao.DAO, tx *transaction.Transaction) error {
...
+			})
+			if i != len(ba) && ba[i].Equals(signer.Account) {
+				return fmt.Errorf("account %s is blocked", signer.Account.StringLE())
 			}
 		}
 	}
+	maxBlockSystemFee := p.GetMaxBlockSystemFeeInternal(d)
+	if maxBlockSystemFee < tx.SystemFee {
+		return fmt.Errorf("transaction's fee can't exceed maximum block system fee %d", maxBlockSystemFee)
+	}
```
Comment (AnnaShaleva) [inline pkg/core/native/policy.go:532]: The comment to `CheckPolicy` should also be updated according to these changes.

File: pkg/core/blockchain.go [hunk 6]
```diff
@@ -1219,26 +1219,26 @@ func (bc *Blockchain) verifyTx(t *transaction.Transaction, block *block.Block) e
...
 	balance := bc.GetUtilityTokenBalance(t.Sender())
 	need := t.SystemFee + t.NetworkFee
 	if balance.Cmp(big.NewInt(need)) < 0 {
-		return errors.Errorf("insufficient funds: balance is %v, need: %v", balance, need)
+		return fmt.Errorf("insufficient funds: balance is %v, need: %v", balance, need)
 	}
 	size := io.GetVarSize(t)
 	if size > transaction.MaxTransactionSize {
-		return errors.Errorf("invalid transaction size = %d. It shoud be less then MaxTransactionSize = %d", io.GetVarSize(t), transaction.MaxTransactionSize)
+		return fmt.Errorf("invalid transaction size = %d. It shoud be less then MaxTransactionSize = %d", io.GetVarSize(t), transaction.MaxTransactionSize)
```
[the remark this answers is not in the document]
Comment (roman-khimov) [reply pkg/core/blockchain.go:1220]: errors.Errorf -> fmt.Errorf was done automatically, so messages were not improved there, but thanks for comments, I'll improve them where possible.

**Pull request merged**